django/contrib/postgres/signals.py

```diff
@@ -9,6 +9,12 @@ def register_type_handlers(connection, **kwargs):
 
     try:
         register_hstore(connection.connection, globally=True)
+        if ('django.contrib.postgresql.fields.HStoreField' not in
+                connection.introspection.data_types_reverse.values()):
```

Comment:
shouldn't this logic be the other way - if nothing has registered itself for the `oid`, register ourselves? potentially some *other* implementation/subclass of `HStoreField` has already inserted itself into `data_types_reverse` with the `oid` key (similar to my comment in https://github.com/django/django/pull/9049/files#r137933799 ). I guess you're trying to avoid the query here though if this function gets called more than once?